src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,276 @@
+                    {
+                        outputDict[languageWorkerFolder] = languageWorkerVersionPath;
+                        break;
+                    }
+                }
+            }
+
+            return outputDict;
+        }
+
+        private Dictionary<string, string> GetWorkerConfigsFromWithinHost(string fallbackPath, string workerRuntime, Dictionary<string, string> outputDict)
+        {
+            if (Directory.Exists(fallbackPath))
+            {
+                foreach (var workerPath in Directory.EnumerateDirectories(fallbackPath))
+                {
+                    string workerDir = Path.GetFileName(workerPath).ToLower();
+
+                    if (outputDict.ContainsKey(workerDir) ||
+                    (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                    workerRuntime is not null &&
+                    !workerRuntime.Equals(workerDir, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        continue;
+                    }
+
+                    string workerConfigPath = Path.Combine(workerPath, RpcWorkerConstants.WorkerConfigFileName);
+                    if (File.Exists(workerConfigPath))
+                    {
+                        outputDict[workerDir] = workerPath;
+                    }
+
+                    if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                        workerRuntime is not null &&
+                        outputDict.ContainsKey(workerRuntime))
+                    {
+                        break;
+                    }
+                }
+            }
+
+            return outputDict;
+        }
+
+        private static IEnumerable<Version> GetWorkerVersionsDescending(IEnumerable<string> workerVersions)
+        {
+            if (!workerVersions.Any())
+            {
+                return Enumerable.Empty<Version>();
+            }
+
+            var versions = new List<Version>();
+
+            foreach (var workerVersion in workerVersions)
+            {
+                string versionDir = Path.GetFileName(workerVersion);
+                string formattedVersion = FormatVersion(versionDir);
+
+                if (Version.TryParse(formattedVersion, out Version version))
+                {
+                    versions.Add(version);
+                }
+            }
+
+            return versions.OrderDescending();
+        }
+
+        private static string FormatVersion(string version)
+        {
+            if (!version.Contains('.'))
+            {
+                version = version + ".0"; // Handle versions like '1' as '1.0'
+            }
+
+            return version;
+        }
+
+        private bool IsWorkerCompatibleWithHost(string workerDir)
+        {
+            string workerConfigPath = Path.Combine(workerDir, RpcWorkerConstants.WorkerConfigFileName);
+            if (!File.Exists(workerConfigPath))
```

Comment:
did we determine that this is the "sentinel" file that indicates installation is complete? or should we also check some "worker.installed" file that we will drop as a final step during unzipping? We need to prevent races where there's a partial unzipping and we try to run the worker anyway